Java/core/src/main/java/com/amazon/randomcutforest/sampler/CompactSampler.java

```diff
@@ -271,32 +298,30 @@ public void addSample(Integer pointIndex, float weight, long sequenceIndex) {
             } else
                 break;
         }
+        acceptPointState = null;
     }
 
     @Override
-    public List<Sequential<Integer>> getSequentialSamples() {
-        checkState(storeSequenceIndexesEnabled, "to call this method storeSequenceIndexesEnabled must be true");
-        List<Sequential<Integer>> result = new ArrayList<>();
-        for (int i = 0; i < size; i++) {
-            result.add(new Sequential<>(pointIndex[i], weight[i], sequenceIndex[i]));
-        }
-        return result;
+    public List<Sampled<Integer>> getSamples() {
+        return streamSamples().collect(Collectors.toList());
     }
 
-    @Override
     public List<Weighted<Integer>> getWeightedSamples() {
-        List<Weighted<Integer>> result = new ArrayList<>();
-        for (int i = 0; i < size; i++) {
-            result.add(new Weighted<>(pointIndex[i], weight[i]));
-        }
-        return result;
+        return streamSamples().collect(Collectors.toList());
+    }
+
+    private Stream<Weighted<Integer>> streamSamples() {
+        return IntStream.range(0, size).mapToObj(i -> {
+            long index = sequenceIndex != null ? sequenceIndex[i] : 0L;
```

Comment:
This needs to be documented (the fact that we set sequence index to 0 in the returned values if store-sequence-index is not enabled.